pydarn/io/borealis/borealis_convert.py

```diff
@@ -202,9 +218,22 @@ def __init__(self, borealis_filename: str, borealis_filetype: str,
         self.borealis_records = self.records
         self.sdarn_filename = sdarn_filename
         self.borealis_filename = self.filename
-        self._borealis_slice_id = borealis_slice_id
+
+        try:
+            first_key = list(self.records.keys())[0]
+            self._borealis_slice_id = self.records[first_key]['slice_id']
+        except KeyError as e:
+            if borealis_slice_id is not None:
+                self._borealis_slice_id = int(borealis_slice_id)
+            else:
+                raise borealis_exceptions.BorealisStructureError(
+                    'The slice_id could not be found in the file: Borealis '
+                    'files produced before Borealis v0.5 must provide the '
+                    'slice_id value to the BorealisConvert class.') from e
+
         self._sdarn_dmap_records = {}
         self._sdarn_dict = {}
+        self._scaling_factor = scaling_factor
```

Comment:
Was doing some testing and came across an error that was not very clear for the user and would be solved by this suggestion!

```suggestion
        self._scaling_factor = int(scaling_factor)
```